File: Character_Stuffing.java
import java.util.Scanner;

public class Character_Stuffing {
    public static void main(String[] args) {
        Scanner sc = new Scanner(System.in);

        // Input Section
        System.out.print("Enter the data to be stuffed: ");
        String input = sc.nextLine();

        System.out.print("Enter the starting delimiter character: ");
        char startDelim = sc.next().charAt(0);

        System.out.print("Enter the ending delimiter character: ");
        char endDelim = sc.next().charAt(0);

        // Prepare doubled delimiters
        String doubleStart = "" + startDelim + startDelim;
        String doubleEnd = "" + endDelim + endDelim;

        // Start framing with starting delimiter
        StringBuilder stuffed = new StringBuilder();
        stuffed.append(doubleStart);

        // Stuffing logic
        for (int i = 0; i < input.length(); i++) {
            char ch = input.charAt(i);
            if (ch == startDelim) {
                stuffed.append(doubleStart); // Stuff start delimiter
            } else if (ch == endDelim) {
                stuffed.append(doubleEnd); // Stuff end delimiter
            } else {
                stuffed.append(ch);
            }
        }

        // Add ending delimiter
        stuffed.append(doubleEnd);

        // Output
        System.out.println("Data after character stuffing: " + stuffed.toString());

        sc.close();
    }
}

File: Character_Stuffing.py
def character_stuffing():
    # Input Section
    data = input("Enter the data to be stuffed: ")

    start_delim = input("Enter the starting delimiter character: ")[0]
    end_delim = input("Enter the ending delimiter character: ")[0]

    # Prepare doubled delimiters
    double_start = start_delim * 2
    double_end = end_delim * 2

    # Start framing with starting delimiter
    stuffed = double_start

    # Stuffing logic
    for ch in data:
        if ch == start_delim:
            stuffed += double_start
        elif ch == end_delim:
            stuffed += double_end
        else:
            stuffed += ch

    # Add ending delimiter
    stuffed += double_end

    # Output
    print("Data after character stuffing:", stuffed)


# Run the function
if __name__ == "__main__":
    character_stuffing()
